decidim-core/app/controllers/decidim/participatory_processes_controller.rb

```diff
@@ -6,15 +6,14 @@ module Decidim
   # A controller that holds the logic to show ParticipatoryProcesses in a
   # public layout.
   class ParticipatoryProcessesController < ApplicationController
-    include NeedsParticipatoryProcess
-
     layout "layouts/decidim/participatory_process", only: [:show]
 
-    skip_after_action :verify_participatory_process, only: [:index]
+    before_action -> { extend(NeedsParticipatoryProcess) }, only: [:show]
```

Comment:
This is a very good observation. To my understanding, this extends the controller's instance singleton class, so this won't affect the real class. If it doesn't work as I expect, this totally ruins my plans for the `assemblies` feature... :disappointed:.

Let me give it some more testing. I guess I should try this without class reloading enabled, right?